In [ ]:
# HYPOTHESIS TESTING

from scipy import stats
import math

# Given
x_bar = 3050.0
mu_0 = 1000 + 5 * 600
sigma = 5 * 25
n = 25
se = sigma / math.sqrt(n)

# Test statistic (z)
z = (x_bar - mu_0) / se

# Critical value (one-tailed, alpha = 0.05)
alpha = 0.05
z_crit = stats.norm.ppf(1 - alpha)

# p-value (upper-tail for Ha: mu > mu0)
p_value = 1 - stats.norm.cdf(z)

# Decision
decision = "Reject H0" if z > z_crit else "Do not reject H0"

# Print results
print("Hypothesis Test (Z) — Alternative hypothesis: mu > mu0")
print("-----------------------------------")
print(f"Sample mean (x̄)                   : {x_bar}")
print(f"Theoretical mean (μ0)             : {mu_0}")
print(f"Sigma (σ)                         : {sigma}, SE: {se:.2f}")
print(f"z-statistic                       : {z:.4f}")
print(f"z-critical (α={alpha}, one-tailed)   : {z_crit:.6f}")
print(f"p-value (upper-tail)              : {p_value:.6e}")
print(f"Decision                          : {decision}")


Hypothesis Test (Z) — Alternative hypothesis: mu > mu0
-----------------------------------
Sample mean (x̄)                   : 3050.0
Theoretical mean (μ0)             : 4000
Sigma (σ)                         : 125, SE: 25.00
z-statistic                       : -38.0000
z-critical (α=0.05, one-tailed)   : 1.644854
p-value (upper-tail)              : 1.000000e+00
Decision                          : Do not reject H0


**Conclusion**

Based on the one-sample z-test at the 5% significance level, the calculated test statistic was z = -38.00, while the critical value for an upper-tailed test is z = 1.645. Since the test statistic does not exceed the critical value and the p-value is very large (≈ 1.0), we fail to reject the null hypothesis.

Therefore, there is no statistical evidence to support the restaurant owners’ claim that the actual weekly operating costs are higher than the theoretical model predicts. In fact, the sample mean weekly cost (3050) is substantially lower than the model’s expected cost (4000).

In [ ]:
# CHI-SQUARE TESTING

import numpy as np
from scipy.stats import chi2_contingency

# Contingency table (Observed frequencies)
# Rows = Satisfaction levels
# Columns = [Smart Thermostat, Smart Light]
observed = np.array([
    [50, 70],   # Very Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
])

# Perform Chi-Square Test for Independence
chi2, p, dof, expected = chi2_contingency(observed)

# Print results
print("Chi-Square Test for Independence")
print("--------------------------------")
print(f"Chi-square statistic       : {chi2:.4f}")
print(f"Degrees of freedom         : {dof}")
print(f"Expected frequencies       :\n", expected)
print(f"P-value                    : {p:.4f}")

# Decision at alpha = 0.05
alpha = 0.05
if p < alpha:
    print("\nConclusion: Reject H0 (There is a significant association).")
else:
    print("\nConclusion: Fail to reject H0 (No significant association).")


Chi-Square Test for Independence
--------------------------------
Chi-square statistic       : 5.6382
Degrees of freedom         : 4
Expected frequencies       :
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]
P-value                    : 0.2278

Conclusion: Fail to reject H0 (No significant association).


In [ ]:
from scipy.stats import chi2

# Significance level
alpha = 0.05

# Degrees of freedom: (rows-1)*(columns-1)
df = (5 - 1) * (2 - 1)  # 4

# Critical value for upper tail
chi2_crit = chi2.ppf(1 - alpha, df)

print(f"Critical value (chi-square, alpha={alpha}, df={df}): {chi2_crit:.4f}")


Critical value (chi-square, alpha=0.05, df=4): 9.4877


**Conclusion:**

The Chi-Square test yielded a statistic of 5.64 (critical value = 9.488, p-value = 0.228). Since the statistic is less than the critical value and the p-value is greater than 0.05, we fail to reject the null hypothesis. Therefore, there is no significant association between device type and customer satisfaction.